src/twisted/spread/jelly.py

```diff
@@ -538,9 +533,9 @@ def jelly(self, obj):
                     sxp.append(dictionary_atom)
                     for key, val in obj.items():
                         sxp.append([self.jelly(key), self.jelly(val)])
-                elif objType in _SetTypes:
+                elif objType is set:
                     sxp.extend(self._jellyIterable(set_atom, obj))
-                elif objType in _ImmutableSetTypes:
+                elif objType is frozenset:
```

Comment:
@wiml this is a slightly breaking change:
```
class TypeEqAllMeta(type):
    def __eq__(self, other):
        return True

class TypeEqAll(metaclass=TypeEqAllMeta):
    pass


v = TypeEqAll()

print(f"{type(v) in (set,)=} {type(v) in (frozenset,)=}") # type(v) in (set,)=True type(v) in (frozenset,)=True
print(f"{type(v) is set=} {type(v) is frozenset=}")  # type(v) is set=False type(v) is frozenset=False
```

which is probably fine - but something to be aware of

Reply:
That's a good point I hadn't considered (classes which may be `==` without being `is`). As you say this is unlikely to break anything but I appreciate the feedback.